bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/ItemCommandActionHandler.java

```diff
@@ -102,10 +103,21 @@ public void dispose() {
     public Map<String, Object> execute(Map<String, Object> inputs) {
         String itemName = (String) module.getConfiguration().get(ITEM_NAME);
         String command = (String) module.getConfiguration().get(COMMAND);
-        if (itemName != null && command != null && eventPublisher != null && itemRegistry != null) {
+
+        if (itemName != null && eventPublisher != null && itemRegistry != null) {
             try {
                 Item item = itemRegistry.getItem(itemName);
-                Command commandObj = TypeParser.parseCommand(item.getAcceptedCommandTypes(), command);
+
+                Command commandObj = null;
+                Object cmd = inputs.get(COMMAND);
```

Comment:
is `inputs` always none-null or is this a potential NPE?

Reply:
I can be `null`, but if it is, you will end up in the `else` branch of the following `if`, because `null` is not an instance of `Command`.

Reply:
yes, true for `cmd`. But what about `inputs`?

Reply:
See

https://github.com/eclipse/smarthome/blob/2fd9709b37a63b203395f28b68f5098fbbdecf06/bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/handler/ActionHandler.java#L34-L43

And note that the class is annotated with `@NonNullByDefault`. So the `inputs` in my case are not null if the framework calls the `execute` method.

Reply:
thanks.